engine/schema/src/main/java/com/cloud/vm/dao/ConsoleSessionDaoImpl.java

```diff
@@ -19,11 +19,24 @@
 
 package com.cloud.vm.dao;
 
+import com.cloud.utils.db.SearchBuilder;
+import com.cloud.utils.db.SearchCriteria;
 import com.cloud.vm.ConsoleSessionVO;
 import com.cloud.utils.db.GenericDaoBase;
 
+import java.util.Date;
+import java.util.List;
+
 public class ConsoleSessionDaoImpl extends GenericDaoBase<ConsoleSessionVO, Long> implements ConsoleSessionDao {
 
+    private final SearchBuilder<ConsoleSessionVO> SearchByRemovedDate;
+
+    public ConsoleSessionDaoImpl() {
+        SearchByRemovedDate = createSearchBuilder();
+        SearchByRemovedDate.and("removed", SearchByRemovedDate.entity().getRemoved(), SearchCriteria.Op.NNULL);
```

Comment:
is this an issue that two conditions have the same key "removed" ?
may be better to rename key in line 36 to "removedIsNotNull" (just an example)

Reply:
Thanks